test/grouping.jl

```diff
@@ -1458,4 +1456,42 @@ end
     @test_throws KeyError gd[(a=:A, b=:X)]
 end
 
+@testset "Allow returning DataFrame() or NamedTuple() to drop group" begin
+    for i in typemin(UInt8):typemax(UInt8),
+        er in (DataFrame(), view(DataFrame(ones(2,2)), 2:1, 2:1),
+               view(DataFrame(ones(2,2)), 1:2, 2:1),
+               NamedTuple(), rand(0,0), rand(5,0))
+        df = DataFrame(a = 1:8, x1 = Bool.(collect(bitstring(UInt8(i))) .- '0'))
```

Comment:
The definition of `x1` is really tricky. Can you add a comment explaining what it does? IIUC the idea is to cover all possible sequences?

Reply:
Yes. Unfortunately (probably) this was a most natural way to define it for me. I have changed it to something more normal now.